tools/bridge_inundation.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import errno
+import os
+from timeit import default_timer as timer
+
+import geopandas as gpd
+import pandas as pd
+
+
+def bridge_risk_status(hydrofabric_dir: str, flow_file_dir: str, output_dir: str) -> gpd.GeoDataFrame:
+    """
+    This function detect which bridge points are affected by a specified flow file. The function requires a flow file (expected to follow
+    the schema used by 'inundation_mosaic_wrapper') with data organized by 'feature_id' and 'discharge' in cms. The output includes a geopackage
+    containing bridge points labeled as "threatened", "at risk", or "not at risk" based on forcasted discharge compared to preset discharge
+    ("max_discharge" or "max_discharge75").
+
+    Args:
+        hydrofabric_dir (str):    Path to hydrofabric directory where FIM outputs were written by
+                                    fim_pipeline.
+        flow_file_dir (str):      Path to flow file to be used for inundation.
+                                    feature_ids in flow_file should be present in supplied HUC.
+        output (str):             Path to output geopackage.
+    """
+
+    dir_path = hydrofabric_dir
+    # Check that hydrofabric_dir exists
+    if not os.path.exists(dir_path):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), dir_path)
+
+    # Get the list of all hucs in the directory
+    entries = os.listdir(dir_path)
```

Comment:
There are other folders in the FIM hydrofabric folder (e.g., unit_errors, branch_errors, and logs) that are getting added to the list. You might use `entries = [d for d in os.listdir(dir_path) if re.match(r'^\d{8}$', d)]` to filter just the 8-digit HUC names.